src/analyses/ai.h

```diff
@@ -358,6 +366,21 @@ class ai_baset
   virtual std::unique_ptr<statet> make_temporary_state(const statet &s)=0;
 };
 
+namespace detail // NOLINT
+{
+/// Functor to compare iterators from different collections for equality.
+/// Declared outside of ait so that only a single class is declared, instead
+/// of a separate one for each type supplied as an argument to ait.
+struct location_equalt
+{
+  typedef goto_programt::const_targett locationt;
+  bool operator()(const locationt &a, const locationt &b) const
+  {
+    return &(*a) == &(*b);
+  }
+};
+} // namespace detail
```

Comment:
I'd suggest that this be moved to goto-programs/goto_program_template.h, which already has some related comparison code.